### DISCUSSION TITLE
Cross-compiling

### DISCUSSION THREAD

**Discussion by andy121090** [mitmproxy/mitmproxy#6048] [Q&A]
Hi,

I try to compile the mitmproxy binaries in an OpenWrt architecture.
According to https://openwrt.org/docs/guide-developer/toolchain/crosscompile, one has to set the environment variables `CC` and `LD` during the Make build.
What would be the recommended way to do this in the Python build?

I can create the binaries by following the commands in https://github.com/mitmproxy/mitmproxy/blob/main/.github/workflows/main.yml. However, setting the environment variables seems to have no effect. When I run the executable on the target system, I get the following error:
`./mitmproxy: line 1: syntax error: unexpected ")"`

**Comment by mhils** [comment]
If possible I would try to avoid cross-compiling and compile on the target platform instead. If you absolutely need to cross-compile, I would avoid bringing PyInstaller into the equation and instead focus on building Python wheels for your target platform. [`release/docker/Dockerfile`](https://github.com/mitmproxy/mitmproxy/blob/main/release/docker/Dockerfile) shows how that works with different systems on the same architecture.

**Reply by andy121090** [reply]
Thank you for the answer!
As this comment https://github.com/openwrt/packages/issues/11008#issuecomment-578739169 suggests, I need to compile the version `7.0.4` of mitmproxy on the target platform because the version of cryptography is `3.4.8`.
However, following the build in https://github.com/mitmproxy/mitmproxy/blob/v7.0.4/.github/workflows/main.yml leads to the error: `__main__.BuildError: We're on neither a tag nor a branch - could not establish version`
Am I missing certain environment variables?

**Reply by mhils** [reply]
We provide no support for outdated releases at all, sorry.

**Comment by lostwindsenril** [comment]
Note that Openwrt is based on musl rather than glibc. I'm still working on it.
